test/robustness/overlay/areal_areal/random_integer_grids.cpp

```diff
@@ -0,0 +1,267 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+// Robustness Test
+
+// Copyright (c) 2025 Tinko Bartels, Shenzhen, China.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#define BOOST_GEOMETRY_NO_BOOST_TEST
+
+#include <bitset>
+#include <iostream>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include <boost/program_options.hpp>
+
+#include <boost/geometry/algorithms/difference.hpp>
+#include <boost/geometry/algorithms/union.hpp>
+#include <boost/geometry/algorithms/sym_difference.hpp>
+#include <boost/geometry/algorithms/intersection.hpp>
+#include <boost/geometry/algorithms/is_valid.hpp>
+#include <boost/geometry/io/wkt/write.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+
+#define BOOST_GEOMETRY_TEST_ONLY_ONE_TYPE
+#define BOOST_GEOMETRY_DEFAULT_TEST_TYPE int
+#include <geometry_test_common.hpp>
+
+constexpr int chunk_size = 64;
+using bits = std::vector<std::bitset<chunk_size>>;
+using prng = std::mt19937_64;
+static_assert(sizeof(prng::result_type) >= chunk_size / 8, "PRNG output too narrow.");
+
+namespace bg = boost::geometry;
+
+using point = bg::model::d2::point_xy<int>;
+using box   = bg::model::box<point>;
+using mp    = bg::model::multi_polygon<bg::model::polygon<point>>;
+
+struct grid_settings
+{
+    int width              = 5;
+    int height             = 5;
+    int count              = 1;
+    prng::result_type seed = prng::default_seed;
+};
+
+constexpr int cell_dimension = 2;
+
+std::vector<box> grid_cells(grid_settings const& settings)
+{
+    std::vector<box> out;
```

Comment:
You could reserve width*height, here and below - it's probably a bit faster when testing a lot. Though the bg operations might overshadow the gain, of course.